### PR TITLE
Fixed and refactored crouch states (WIP)

### PR DESCRIPTION
**Pull request by commodare**

### CODE REVIEW

File: Assets/Editor/GroundhogEditorMenu.cs [hunk 1]
```diff
@@ -10,4 +11,70 @@ static void MovePlayerHere()
     {
         GameObject.FindWithTag("Player").transform.position = SceneView.lastActiveSceneView.camera.transform.position;
     }
+
+     [MenuItem("Assets/Create Reversed Clip", false, 14)]
```
Comment (sapphon) [inline Assets/Editor/GroundhogEditorMenu.cs:15]: What does this do?

File: Assets/Editor/GroundhogEditorMenu.cs [hunk 2]
```diff
@@ -10,4 +11,70 @@ static void MovePlayerHere()
...
+        var events = AnimationUtility.GetAnimationEvents(clip);
+        if (events.Length > 0)
+        {
+            for (int i = 0; i < events.Length; i++)
+            {
+                events[i].time = clipLength - events[i].time;
+            }
+            AnimationUtility.SetAnimationEvents(clip, events);
+        }
+        Debug.Log("Animation reversed!");
```
Comment (sapphon) [inline Assets/Editor/GroundhogEditorMenu.cs:62]: ಠ_ಠ

File: Assets/Editor/GroundhogEditorMenu.cs [hunk 3]
```diff
@@ -10,4 +11,70 @@ static void MovePlayerHere()
...
+        var clip = GetSelectedClip();
+ 
+        AssetDatabase.CopyAsset(AssetDatabase.GetAssetPath(Selection.activeObject), copiedFilePath);
+ 
+        clip  = (AnimationClip)AssetDatabase.LoadAssetAtPath(copiedFilePath, typeof(AnimationClip));
+ 
+        if (clip == null)
+            return;
+        float clipLength = clip.length;
+        var curves = AnimationUtility.GetAllCurves(clip, true);
```
Comment (sapphon) [inline Assets/Editor/GroundhogEditorMenu.cs:32]: this line causes a warning, although our product has a few of those for sure

File: Assets/Editor/GroundhogEditorMenu.cs
Comment (sapphon) [inline Assets/Editor/GroundhogEditorMenu.cs:1]: this file in general will need to be refactored but it'll be easier for me to do than explain, so resubmit once crouching works and I'll do it up

**Review by sapphon — CHANGES_REQUESTED** [review]
I can't accept this yet; 2 issues:

- crouching seems to work better, but the character can now only crouch while moving - it's not possible to crouch while standing still
- uncrouch speed is different depending on whether standing still or moving; standing-still seems like the intended speed and uncrouch while moving seems very slow

File: Assets/Editor/GroundhogEditorMenu.cs (hunk 1 above)
Comment (commodare) [reply Assets/Editor/GroundhogEditorMenu.cs:15]: This allows the user to access the function by right-clicking an item in the project window.

File: Assets/Editor/GroundhogEditorMenu.cs
Comment (commodare) [reply Assets/Editor/GroundhogEditorMenu.cs:1]: Roger!

Comment (commodare) [reply Assets/Editor/GroundhogEditorMenu.cs:1]: All done

File: Assets/Editor/GroundhogEditorMenu.cs (hunk 3 above)
Comment (commodare) [reply Assets/Editor/GroundhogEditorMenu.cs:32]: Yeah, Unity doesn't like deprecated methods being used, even though they still work.

**Review by sapphon — APPROVED** [review]
Thank you!

**Pull request merged**